node/builder.go

```diff
@@ -193,170 +320,252 @@ func (nc *Builder) build(ctx context.Context) (*Node, error) {
+	if b.Rewarder == nil {
 		processor = consensus.NewDefaultProcessor()
 	} else {
-		processor = consensus.NewConfiguredProcessor(consensus.NewDefaultMessageValidator(), nc.Rewarder)
+		processor = consensus.NewConfiguredProcessor(consensus.NewDefaultMessageValidator(), b.Rewarder)
 	}
 
-	// set up consensus
-	nodeConsensus := consensus.NewExpected(&ipldCborStore, bs, processor, blkValid, powerTable, genCid, nc.BlockTime, consensus.ElectionMachine{}, consensus.TicketMachine{})
+	// setup block validation
+	// TODO when #2961 is resolved do the needful here.
+	blkValid := consensus.NewDefaultBlockValidator(b.BlockTime, b.Clock)
 
-	// Set up libp2p network
-	// TODO PubSub requires strict message signing, disabled for now
-	// reference issue: #3124
-	fsub, err := libp2pps.NewFloodSub(ctx, peerHost, libp2pps.WithMessageSigning(false))
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to set up network")
-	}
 	// register block validation on floodsub
 	btv := net.NewBlockTopicValidator(blkValid)
-	if err := fsub.RegisterTopicValidator(btv.Topic(network), btv.Validator(), btv.Opts()...); err != nil {
-		return nil, errors.Wrap(err, "failed to register block validator")
+	if err := network.fsub.RegisterTopicValidator(btv.Topic(network.NetworkName), btv.Validator(), btv.Opts()...); err != nil {
+		return ChainSubmodule{}, errors.Wrap(err, "failed to register block validator")
 	}
 
-	backend, err := wallet.NewDSBackend(nc.Repo.WalletDatastore())
+	// set up consensus
+	powerTable := &consensus.MarketView{}
+	nodeConsensus := consensus.NewExpected(blockstore.cborStore, blockstore.Blockstore, processor, blkValid, powerTable, b.genCid, b.BlockTime, consensus.ElectionMachine{}, consensus.TicketMachine{})
+
+	// setup fecher
+	graphsyncNetwork := gsnet.NewFromLibp2pHost(network.host)
+	bridge := ipldbridge.NewIPLDBridge()
+	loader := gsstoreutil.LoaderForBlockstore(blockstore.Blockstore)
+	storer := gsstoreutil.StorerForBlockstore(blockstore.Blockstore)
+	gsync := graphsync.New(ctx, graphsyncNetwork, bridge, loader, storer)
+	fetcher := net.NewGraphSyncFetcher(ctx, gsync, blockstore.Blockstore, blkValid, b.Clock, network.PeerTracker)
+
+	messageStore := chain.NewMessageStore(blockstore.cborStore)
+
+	// only the syncer gets the storage which is online connected
+	chainSyncer := chain.NewSyncer(nodeConsensus, chainStore, messageStore, fetcher, chainStatusReporter, b.Clock)
+
+	chainState := cst.NewChainStateReadWriter(chainStore, messageStore, blockstore.cborStore)
+
+	return ChainSubmodule{
+		// BlockSub: nil,
+		Consensus:    nodeConsensus,
+		ChainReader:  chainStore,
+		MessageStore: messageStore,
+		Syncer:       chainSyncer,
+		PowerTable:   powerTable,
+		// HeaviestTipSetCh: nil,
+		// cancelChainSync: nil,
+		ChainSynced: moresync.NewLatch(1),
+		Fetcher:     fetcher,
+		State:       chainState,
+		validator:   blkValid,
+	}, nil
+}
+
+func (b *Builder) buildBlockMining(ctx context.Context) (BlockMiningSubmodule, error) {
```

Comment:
This is a little weird. The mining submodule's components can't be constructed until mining starts. It makes more sense to me to delay building the `BlockMiningSubmodule` until then too. `buildBlockMinining` isn't really doing what it says it's doing.

Reply:
Same applies for SectorStorage, HelloProtocol, StorageProtocol, RetrievalProtocol and FaultSlasher. Constructing the submodule container and then building the components later by accessing the submodule through the node seems counter to the spirit of this refactor.

Reply:
This PR is being cut in half, the next half PR will address cleaning up the node.Start function. 
A lot of those submodules *currently* (as in pre-refactor) seems to exists only during start, some get initialized during start but could have been initialized during construction.

It will start making sense in a couple more PRs. For now things are gravitating towards were the will end up without pushing a 5K LoC PR.